File: scripts/screenshot.js
const { firefox } = require("playwright");

(async () => {
  const browser = await firefox.launch();

  // ── Mobile 390×844 ────────────────────────────────────────────────────────
  const mobile = await browser.newPage({ viewport: { width: 390, height: 844 } });
  await mobile.goto("http://127.0.0.1:8765/");
  await mobile.waitForTimeout(3000);

  await mobile.screenshot({ path: "screenshots/04-mobile-initial.png" });
  console.log("04-mobile-initial.png");

  await mobile.click("#toggle-nodes-drawer");
  await mobile.waitForTimeout(500);
  await mobile.screenshot({ path: "screenshots/05-mobile-drawer-closed.png" });
  console.log("05-mobile-drawer-closed.png");

  await mobile.click("#toggle-traffic-drawer");
  await mobile.waitForTimeout(500);
  await mobile.screenshot({ path: "screenshots/06-mobile-traffic-open.png" });
  console.log("06-mobile-traffic-open.png");

  await mobile.close();

  // Desktop: verify traffic drawer
  const desktop = await browser.newPage({ viewport: { width: 1440, height: 900 } });
  await desktop.goto("http://127.0.0.1:8765/");
  await desktop.waitForTimeout(3000);
  await desktop.click("#toggle-traffic-drawer");
  await desktop.waitForTimeout(600);
  await desktop.screenshot({ path: "screenshots/07-desktop-traffic-alone.png" });
  console.log("07-desktop-traffic-alone.png");

  await desktop.close();
  await browser.close();
  console.log("\nDone.");
})();

File: scripts/screenshot.py
"""Headless browser screenshot validation for meshradar UI."""
from playwright.sync_api import sync_playwright
import os

BASE_URL = "http://127.0.0.1:8765"
OUT_DIR = "screenshots"
os.makedirs(OUT_DIR, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch()

    # ── Desktop 1440×900 ──────────────────────────────────────────────────────
    page = browser.new_page(viewport={"width": 1440, "height": 900})
    page.goto(BASE_URL)
    page.wait_for_load_state("networkidle")
    page.wait_for_timeout(1200)

    # 01: initial state
    page.screenshot(path=f"{OUT_DIR}/01-desktop-initial.png")
    print("01-desktop-initial.png")

    # 02: drawer closed
    page.click("#toggle-nodes-drawer")
    page.wait_for_timeout(400)
    page.screenshot(path=f"{OUT_DIR}/02-desktop-drawer-closed.png")
    print("02-desktop-drawer-closed.png")

    # 03: drawer open + traffic open
    page.click("#toggle-nodes-drawer")
    page.wait_for_timeout(300)
    page.click("#toggle-traffic-drawer")
    page.wait_for_timeout(400)
    page.screenshot(path=f"{OUT_DIR}/03-desktop-traffic-open.png")
    print("03-desktop-traffic-open.png")

    page.close()

    # ── Mobile 390×844 ────────────────────────────────────────────────────────
    page = browser.new_page(viewport={"width": 390, "height": 844})
    page.goto(BASE_URL)
    page.wait_for_load_state("networkidle")
    page.wait_for_timeout(1200)

    # 04: mobile initial
    page.screenshot(path=f"{OUT_DIR}/04-mobile-initial.png")
    print("04-mobile-initial.png")

    # 05: mobile with drawer open
    page.click("#toggle-nodes-drawer")
    page.wait_for_timeout(400)
    page.screenshot(path=f"{OUT_DIR}/05-mobile-drawer-open.png")
    print("05-mobile-drawer-open.png")

    page.close()
    browser.close()

print(f"\nAll screenshots saved to {OUT_DIR}/")
